### PR TITLE
[nginx-ingress-controller]: Avoid sync without a reachable master

### LINKED ISSUE #1226: Ingress Controller doesn't handle very well when master is down
I just had to update the box where my Kubernetes master is running.
While the box was rebooting, the Ingress Controller lost all the secrets (TLS), and the HTTP basic auth configuration. In the end, all the Ingress Resources were just listening on 80. No more TLS, no more HTTP Basic Auth (and probably all the other configurations depending on annotations).

When master is not available, it shouldn't really update its nginx.conf. Thoughts ?

### PR DESCRIPTION
**Pull request by aledbf**
fixes #1226

### CODE REVIEW

**Comment by aledbf** [comment]
ping @bprashanth

File: ingress/controllers/nginx/controller.go [hunk 1]
```diff
@@ -421,13 +421,17 @@ func (lbc *loadBalancerController) sync(key string) {
...
-	var cfg *api.ConfigMap
+	cfg := &api.ConfigMap{}
+	var err error
 
-	ns, name, _ := parseNsName(lbc.nxgConfigMap)
-	cfg, err := lbc.getConfigMap(ns, name)
-	if err != nil {
-		glog.V(3).Infof("unexpected error searching configmap %v: %v", lbc.nxgConfigMap, err)
-		cfg = &api.ConfigMap{}
+	if lbc.nxgConfigMap != "" {
```
Comment (bprashanth) [inline ingress/controllers/nginx/controller.go]: please add a comment on when this might be empty in the abnormal case

**Comment by bprashanth** [comment]
LGTM, just comment nit

File: ingress/controllers/nginx/controller.go (hunk 1 above)
Comment (aledbf) [inline ingress/controllers/nginx/controller.go]: done

**Comment by bprashanth** [comment]
Reverting, @aledbf I think we chaned a method signature with the delayerd queue that didn't rebase too well with this change

**Pull request merged**